kernel/Kconfig.obj_core

```diff
@@ -77,7 +77,7 @@ config OBJ_CORE_SEM
 
 config OBJ_CORE_PIPE
 	bool "Integrate pipe into object core framework"
-	default y if PIPES
+	default y
```

Comment:
So pipes are now enabled by default, this might have some footprint implication on those not using the pipe, why does it need to be enabled by default?

Reply:
> why does it need to be enabled by default?

Strictly speaking it deosn't **need** to be. However, similar data structures are typically enabled automatically when `CONFIG_MULTITHREADING` is set. In this case, the pipe implementation and atomic services are the only data structures treated differently.
I assumed enabling it by default would have minimal impact, as the additional pipe structure would only slightly increase build time and should be optimized away during compilation.

Moreover, this approach eliminates the need for an extra Kconfig option to manage mutually exclusive configurations when selecting a pipe implementation during the deprecation process.

Do you prefer an alternative approach to handle this?